frontend/web/components/pages/project-settings/ProjectSettingsPage.tsx

```diff
@@ -0,0 +1,149 @@
+import React, { useEffect } from 'react'
+import PageTitle from 'components/PageTitle'
+import Tabs from 'components/navigation/TabMenu/Tabs'
+import TabItem from 'components/navigation/TabMenu/TabItem'
+import BetaFlag from 'components/BetaFlag'
+import { useGetProjectQuery } from 'common/services/useProject'
+import { useRouteContext } from 'components/providers/RouteContext'
+import Constants from 'common/constants'
+import Utils from 'common/utils/utils'
+import ProjectUsage from 'components/ProjectUsage'
+import EditHealthProvider from 'components/EditHealthProvider'
+import FeatureExport from 'components/import-export/FeatureExport'
+import { GeneralTab } from './tabs/general-tab'
+import { SDKSettingsTab } from './tabs/SDKSettingsTab'
+import { PermissionsTab } from './tabs/PermissionsTab'
+import { CustomFieldsTab } from './tabs/CustomFieldsTab'
+import { ImportTab } from './tabs/ImportTab'
+
+const ProjectSettingsPage = () => {
+  const { environmentId, projectId } = useRouteContext()
+  const {
+    data: project,
+    isLoading,
+    isUninitialized,
+  } = useGetProjectQuery({ id: String(projectId) }, { skip: !projectId })
+
+  useEffect(() => {
+    API.trackPage(Constants.pages.PROJECT_SETTINGS)
+  }, [])
+
+  // Show loader only on initial load (not during refetches from mutations)
+  const isInitialLoading = isUninitialized || (isLoading && !project)
+
+  if (isInitialLoading) {
+    return (
+      <div className='app-container container'>
+        <PageTitle title='Project Settings' />
+        <div className='text-center'>
+          <Loader />
+        </div>
+      </div>
+    )
+  }
+
+  const hasEnvironments = !!project?.environments?.length
+  const hasFeatureHealth = Utils.getFlagsmithHasFeature('feature_health')
+
+  // Derive organisationId from project data (not available in route params)
+  const organisationId = project?.organisation
+
+  if (!project || !projectId || !organisationId) {
```

Comment:
NIT: Do you mind keeping the early return just before the final render to have them together?